### PR TITLE
test(threading): Add spec for `ThreadPoolExecutor`

### PR DESCRIPTION
**Pull request by gggritso**
`ThreadPoolExecutor` also obeys hub propagation, but there wasn't a spec for it. This PR adds a bit more coverage.

### CODE REVIEW

File: tests/integrations/threading/test_threading.py [hunk 1]
```diff
@@ -73,6 +75,42 @@ def stage2():
...
+    if propagate_hub:
+        assert len(events) == 1
+        (event,) = events
+        assert event["spans"][0]["trace_id"] == event["spans"][1]["trace_id"]
+        assert event["spans"][1]["trace_id"] == event["spans"][2]["trace_id"]
+        assert event["spans"][2]["trace_id"] == event["spans"][3]["trace_id"]
+        assert event["spans"][3]["trace_id"] == event["spans"][0]["trace_id"]
+    else:
+        (event,) = events
+        assert len(event["spans"]) == 0
```
Comment (gggritso) [inline tests/integrations/threading/test_threading.py:111]: This part seems suspicious. Is this actual behaviour, or did I fail to set up the test correctly?

Comment (gggritso) [reply tests/integrations/threading/test_threading.py:111]: 👍🏻 makes sense

**Review by sl0thentr0py — APPROVED** [review]

**Comment by sl0thentr0py** [comment]
will fix the 2.7 stuff

**Comment by sl0thentr0py** [comment]
alright that works too

**Comment by gggritso** [comment]
@sl0thentr0py I keep playing whack-a-mole with old Python versions 😅 I copy-pasted from other specs for now, but some stuff is still failing. What's the right way to do things like `import concurrent`? Should I do it inside the spec instead of top-of-file?

**Comment by sl0thentr0py** [comment]
pushed fix, we always do an `except ImportError` see last commit

**Comment by sl0thentr0py** [comment]
waddafaq, looking

**Comment by sl0thentr0py** [comment]
obviously it works on my machine

**Comment by gggritso** [comment]
@antonpirker whatever you did, it seems like it worked? Was the test failure just a weird flake of some kind?

**Comment by gggritso** [comment]
@sl0thentr0py @antonpirker what do you think, is this PR good to merge? Is it even worth it?

**Comment by antonpirker** [comment]
Hey @gggritso 
Sorry for the late reply, I was on vacation. 
More coverage is always good, so I will merge this.

**Comment by gggritso** [comment]
@antonpirker awesome, thanks!

**Pull request merged**